### PR TITLE
additional tools for visualization and knowledge graph

### PR DESCRIPTION
**Pull request by hasanp87**
*Issue #, if available:*

*Description of changes:*
2 additional tool examples
1. Visualization with matplotlib after SQL query
2. Query a knowledge graph in Amazon Neptune DB

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: function-calling/multi-tool/create_athena_catalog.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import boto3
+
+bucket = 'bucketname' ### REPLACE WITH YOUR AMAZON S3 BUCKET
+
+### Setup boto3 client for Athena
+athena = boto3.client('athena', region_name='us-east-1')
```
Comment (mttanke) [inline function-calling/multi-tool/create_athena_catalog.py:6]: make region a variable

File: function-calling/multi-tool/create_athena_catalog.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        LOCATION 's3://{bucket}/acme/'
+    ''',
+    ResultConfiguration={
+        'OutputLocation': f's3://{bucket}/athena/'
+    }
+)
+
+### Insert synthetic example data
+transactions_data = [
+    (1, 'Tom Hanks', '2023-01-01', 500.00),
```
Comment (mttanke) [inline function-calling/multi-tool/create_athena_catalog.py:37]: is this data generated? do we have permissions to use it?

**Review by mttanke — CHANGES_REQUESTED** [review]
left some comments

File: function-calling/knowledge-graph/neptune_graph_sparql_tool.ipynb [hunk 3]
```diff
@@ -0,0 +1,1093 @@
...
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "3ea1471f",
+   "metadata": {},
+   "source": [
+    "## Query Neptune Graph with SPARQL in Natural Language Using Function Calling\n",
+    "\n",
+    "This notebook will walk you through the steps to create a tool that can query Amazon Neptune database via SPARQL using function calling in Bedrock Converse API. The tool will be invoked when a user wants to get object litereal information from Amazon Neptune database for a specific object id. LLM(Large Language Model) will pass the required input parameter from the user's question the function *get_objectLiteral* and the fucntion will execute the SPARQL query to fetch the results. At the end LLM(Large Language Model) will give the final response to the user including the query results.\n",
+    "We will use the Human Disease Ontology as a dataset to load to Amazon Neptune https://www.ebi.ac.uk/ols4/ontologies/doid\n",
```
[the remark this answers is not in the document]
Comment (berilunay) [reply function-calling/knowledge-graph/neptune_graph_sparql_tool.ipynb:11]: Hi, it was opensource from this website: https://www.ebi.ac.uk/ols4/ontologies/doid. For which department should I open the ticket to get it approved?

File: function-calling/knowledge-graph/neptune_graph_sparql_tool.ipynb [hunk 4]
```diff
@@ -0,0 +1,1093 @@
...
+    "\n",
+    "2.In the security group of Neptune Database, allow TCP connections from the port 8182(default port of the instance) through the subnet CIDR range where the database instance resides.\n",
+    "\n",
+    "3.This notebook will run as neptune notebook. During the configuration of database cluster allow the instance to create notebook.\n",
+    "\n",
+    "4.After the notebook is being created you can see it in the sagemaker console, under notebooks. The default IAM role of the notebook instance, by default, has list and read permissions for S3 in the policy. Edit the policy and add \"s3:PutObject\" in the actions and add your S3 bucket ARN in the Resources section of the IAM policy. In order to access Amazon Bedrock, attach *AmazonBedrockFullAccess* policy to the same role.\n",
+    "\n",
+    "After you complete all the steps you should have\n",
+    "* Amazon Neptune database instance (Follow [creating an new neptune DB cluster](https://docs.aws.amazon.com/neptune/latest/userguide/get-started-create-cluster.html) or create manually from the AWS Console). You need to set up the cluster within VPC!\n",
```
[the remark this answers is not in the document]
Comment (berilunay) [reply function-calling/knowledge-graph/neptune_graph_sparql_tool.ipynb:37]: I can add a code for attaching adding those 2 policy via python if that works?

File: function-calling/multi-tool/create_athena_catalog.py (hunk 2 above)
Comment (berilunay) [reply function-calling/multi-tool/create_athena_catalog.py:37]: I found it from https://github.com/aws-samples/amazon-bedrock-samples/blob/main/function-calling/create_athena_catalog.py

File: function-calling/knowledge-graph/neptune_graph_sparql_tool.ipynb (hunk 3 above)
[the remark this answers is not in the document]
Comment (hasanp87) [reply function-calling/knowledge-graph/neptune_graph_sparql_tool.ipynb:11]: The dataset is public available under https://creativecommons.org/publicdomain/zero/1.0/ , we can add this to README and I believe no further approvals are required @berilunay

**Pull request merged**